app/assets/src/components/PipelineSampleReport.jsx

```diff
@@ -1034,57 +1034,42 @@ class PipelineSampleReport extends React.Component {
     return category_lowercase;
   }
 
-  render_name(tax_info, report_details, parent) {
-    let tax_scientific_name = tax_info["name"];
+  render_name(tax_info, report_details, parent, openTaxonModal) {
     let tax_common_name = tax_info["common_name"];
-    let taxonName;
+    const taxonName = getTaxonName(tax_info, this.state.name_type);
     let taxonNameDisplay;
```

Comment:
Maybe add `= <span>{taxonName}</span>`; and remove all the elses below.
Allows you to merge he ifs too.